shell/common/api/electron_api_native_image_win.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/common/api/electron_api_native_image.h"
+
+#include <windows.h>  // NOLINT(build/include_order)
+
+#include <thumbcache.h>  // NOLINT(build/include_order)
+#include <wrl/client.h>  // NOLINT(build/include_order)
+
+#include <string>  // NOLINT(build/include_order)
+#include <vector>  // NOLINT(build/include_order)
+
+#include "shell/common/gin_converters/image_converter.h"
+#include "shell/common/gin_helper/promise.h"
+#include "shell/common/skia_util.h"
+#include "ui/gfx/icon_util.h"
+
+namespace electron {
+
+namespace api {
+
+// static
+v8::Local<v8::Promise> NativeImage::CreateThumbnailFromPath(
+    v8::Isolate* isolate,
+    const base::FilePath& path,
+    const gfx::Size& size) {
+  gin_helper::Promise<gfx::Image> promise(isolate);
+  v8::Local<v8::Promise> handle = promise.GetHandle();
+  HRESULT hr;
+
+  if (size.IsEmpty()) {
```

Comment:
IsEmpty() is terser than the previous "width <= 0 ... height <= 0" checks but this revision lets negative sizes get by, e.g. `maxSize { width: -1, height: -1 }`. Changing that to `isEmpty()` isn't necessarily a blocker if you *intend* to let those values through, but since nothing's mentioned in the new docs, I thought I'd ask if you meant to let those through